wire-ios-data-model/Source/ManagedObjectContext/Migration/2.113-2.114/OneOnOneConversationMigrationAction.swift

```diff
@@ -0,0 +1,101 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+final class OneOnOneConversationMigrationAction: CoreDataMigrationAction {
+    let batchSize = 200
+
+    override func execute(in context: NSManagedObjectContext) throws {
+        let request = NSFetchRequest<NSManagedObject>(entityName: ZMUser.entityName())
+        request.fetchBatchSize = batchSize
+        let users = try context.fetch(request)
+
+        for user in users {
+            if let connection = user.value(forKey: "connection") as? NSManagedObject {
+                migrateConnectionOneOnOne(user: user, connection: connection)
+            } else {
+                migrateTeamOneOnOne(user: user, context: context)
+            }
+        }
+    }
+
+    private func migrateConnectionOneOnOne(user: NSManagedObject, connection: NSManagedObject) {
+        guard
+            let conversation = connection.value(forKey: "conversation") as? NSManagedObject
+        else {
+            return
+        }
+
+        user.setValue(conversation, forKey: "oneOnOneConversation")
+    }
+
+    private func migrateTeamOneOnOne(user: NSManagedObject, context: NSManagedObjectContext) {
+        let sessionRequest = NSFetchRequest<NSManagedObject>(entityName: ZMSession.entityName())
+        let result = try? context.fetch(sessionRequest)
```

Comment:
What about errors here and later in this function, at least we should have some logs.
Maybe we can have a wrapper around that catches and logs?